packages/aws-cdk/test/api/logs/cloudwatch-logs.test.ts

```diff
@@ -0,0 +1,287 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import { CloudFormation } from 'aws-sdk';
+import { registerCloudWatchLogGroups } from '../../../lib/api/logs/cloudwatch-logs';
+import { testStack, TestStackArtifact } from '../../util';
+import { MockSdkProvider } from '../../util/mock-sdk';
+
+let logsMockSdkProvider: LogsMockSdkProvider;
+let mockGetEndpointSuffix: () => string;
+
+beforeEach(() => {
+  jest.useFakeTimers();
+  logsMockSdkProvider = new LogsMockSdkProvider();
+  mockGetEndpointSuffix = jest.fn(() => 'amazonaws.com');
+  logsMockSdkProvider.stubGetEndpointSuffix(mockGetEndpointSuffix);
+  // clear the array
+  currentCfnStackResources.splice(0);
+});
+afterAll(() => {
+  jest.useRealTimers();
+});
+
+test('add log groups from lambda function', async () => {
```

Comment:
General comment about all of these tests:

1. Can we see the template first, before we assign physical names to the resources in it with `pushStackResourceSummaries()`?
2. Can we have an empty line between the Given, When, Then parts of the test?

This comment applies to all tests in this file.

Reply:
Updated. Let me know if it's easier to read now.